File: OSX Examples/Objective-C/Sequences/Sequences/ViewController.m
//
//  ViewController.m
//  Sequences
//
//  Created by Aurelius Prochazka on 2/13/15.
//  Copyright (c) 2015 AudioKit. All rights reserved.
//

#import "ViewController.h"
#import "SequencesConductor.h"
#import "AKTools.h"

@implementation ViewController
{
    IBOutlet NSTextField *durationValue;
    IBOutlet NSSlider *durationSlider;
    
    SequencesConductor *conductor;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
     conductor = [[SequencesConductor alloc] init];
}



- (float)getDuration {
    return [AKTools scaleValueFromSlider:durationSlider minimum:0.05 maximum:0.2];
}

- (IBAction)playPhrase:(id)sender
{
    [conductor playPhraseOfNotesOfDuration:[self getDuration]];
}

- (IBAction)playSequenceOfNoteProperties:(id)sender
{
    [conductor playSequenceOfNotePropertiesOfDuration:[self getDuration]];
}

- (IBAction)playSequenceOfInstrumentProperties:(id)sender
{
    [conductor playSequenceOfInstrumentPropertiesOfDuration:[self getDuration]];
}

- (IBAction)moveDurationSlider:(id)sender
{
    float duration  = [AKTools scaleValueFromSlider:durationSlider minimum:0.05 maximum:0.2];
    [durationValue setStringValue:[NSString stringWithFormat:@"%g", duration]];
}


@end

File: iOS Examples/Swift/Sequences/Sequences/ViewController.swift
//
//  ViewController.swift
//  Sequences
//
//  Created by Aurelius Prochazka on 6/30/14.
//  Copyright (c) 2014 Aurelius Prochazka. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var durationValue: UILabel!
    @IBOutlet weak var durationSlider: UISlider!
    
    let conductor = SequencesConductor()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    func getDuration() -> Float {
        return AKTools.scaleValueFromSlider(durationSlider, minimum: 0.05, maximum: 0.2)
    }
    
    @IBAction func playPhrase(sender: UIButton) {
        conductor.playPhraseOfNotesOfDuration(getDuration())
    }
    
    @IBAction func playSequenceOfNoteProperties(sender: UIButton) {
        conductor.playSequenceOfNotePropertiesOfDuration(getDuration())
    }
    
    @IBAction func playSequenceOfInstrumentProperties(sender: UIButton) {
        conductor.playSequenceOfInstrumentPropertiesOfDuration(getDuration())
    }
    
    @IBAction func moveDurationSlider(sender: UISlider) {
        let duration = AKTools.scaleValueFromSlider(durationSlider, minimum: 0.05, maximum: 0.2)
        durationValue.text = String(format: "%g", duration)
    }
}

